### DISCUSSION TITLE
Kathara container depends on

### DISCUSSION THREAD

**Discussion by MrStaf** [Kidel/Kathara#240] [Q&A] [answered]
Hello everyone. 
I have a problem and would like to know if you have a solution. 
I have two containers that I deploy using Kathara (great tool!), a Server container and a Client.
The problem is that the client depends on the server, and it should not start while the server startup-script is running. 
Is there an equivalent to 'depends_on' found in docker compose?
I've thought of adding a delay to the startup, but I'm not sure this is the best method.
Can you think of a solution?
Thanks in advance!

**Comment by lorenzo93** [comment, marked as answer]
HI @MrStaf ,

we have the `lab.dep` file to declare dependencies to Kathará labs.
[Here](https://www.kathara.org/man-pages/kathara-lab.dep.5.html) you can find the documentation about the `lab.dep` file.
The only drawback about using the `lab.dep` is that the startup will be handled monothread instead of the default multithreaded approach.

Let me know if you have any other questions :)

**Reply by Skazza94** [reply]
In your scenario: if the client device is called `client` and the server is called `server`, the `lab.dep` should be the following:

```
client: server
```

Meaning that `client` will start after `server` is fully started. As @lorenzo93 pointed out, using `lab.dep` will switch the lab deployment to single thread. :smile:

**Reply by MrStaf** [reply]
Thanks a lot! 
But I don't know if it's the service running on the client that's random but it works when I put it in client.startup, whereas it doesn't work when I launch it from the Dockerfile with CMD, it crashes on startup, so it's not very easy to debug :)
I'm not sure what's going on, but the `lab.dep` has helped me!
